File: SensorProject/app/src/main/java/com/example/iosdev/sensorproject/ActivityPagerAdapter.java
package com.example.iosdev.sensorproject;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

/**
 * Created by Antti on 27.9.2016.
 */

public class ActivityPagerAdapter extends FragmentPagerAdapter{
    public ActivityPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return StepFragment.newInstance(position + 1);
        }else {
            return SpeedFragment.newInstance(position + 1);
        }
    }

    @Override
    public int getCount() {
        return 2;
    }
    @Override
    public CharSequence getPageTitle(int position) {
        if (position == 0) {
            return "Steps";
        }else{
            return "Speed";
        }
    }
}

File: SensorProject/app/src/main/java/com/example/iosdev/sensorproject/ActivityPagerAdapter.kt
package com.example.iosdev.sensorproject

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Created by Antti on 27.9.2016.
 */

class ActivityPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    val count: Int
        @Override
        get() = 2

    @Override
    fun getItem(position: Int): Fragment {
        return if (position == 0) {
            StepFragment.newInstance(position + 1)
        } else {
            SpeedFragment.newInstance(position + 1)
        }
    }

    @Override
    fun getPageTitle(position: Int): CharSequence {
        return if (position == 0) {
            "Steps"
        } else {
            "Speed"
        }
    }
}
